### DISCUSSION TITLE
unable to push changes

### DISCUSSION THREAD

**Discussion by avinnitesh** [community/community#35010] [Other Feature Feedback, Questions, & Ideas]
Hi ,

I am getting below error while submitting push request:-
git push origin dev
ERROR: Permission to zero-to-mastery/start-here-guidelines.git denied to avinnitesh.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights

Steps followed fork as below:-
git clone git@github.com:zero-to-mastery/start-here-guidelines.git
cd start-here-guidelines
git remote add upstream https://github.com/zero-to-mastery/start-here-guidelines.git
git pull upstream master
</strong>avinash.kant-> git checkout -b dev
Switched to a new branch 'dev'
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git switch dev
M       CONTRIBUTORS.md
Already on 'dev'
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git status
On branch dev
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   CONTRIBUTORS.md

no changes added to commit (use "git add" and/or "git commit -a")
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git add .
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git status
On branch dev
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   CONTRIBUTORS.md

(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git commit -m "added my name"
[dev 53ed2160a] added my name
 Committer: Avinash Kant <avinash.kant@W-DN-ML-AKant.local>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly. Run the
following command and follow the instructions in your editor to edit
your configuration file:

    git config --global --edit

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 1 file changed, 3 insertions(+), 1 deletion(-)
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git push origin dev
ERROR: Permission to zero-to-mastery/start-here-guidelines.git denied to avinnitesh.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git status
On branch dev
nothing to commit, working tree clean
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git push 
fatal: The current branch dev has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin dev

To have this happen automatically for branches without a tracking
upstream, see 'push.autoSetupRemote' in 'git help config'.

(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git push origin dev
ERROR: Permission to zero-to-mastery/start-here-guidelines.git denied to avinnitesh.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights

**Comment by antonblr** [comment]
@avinnitesh - you need to push into your fork instead, and open a PR then. 
Do the following (assuming you're in `start-here-guidelines` folder already and have `dev` branch active):
```bash
# you have this repo forked already
$ git remote add avinnitesh git@github.com:avinnitesh/start-here-guidelines.git
$ git push avinnitesh dev
```

**Comment by avinnitesh** [comment]
Thanks Anton for your help. It is resolved. I had left the hope to resolve
it a few months back. I appreciate it.

Just a quick question next time when I am forking a new repository.  Do i
have to do remote add similar like up r normal command like below:-
git clone ***@***.***:zero-to-mastery/start-here-guidelines.git
cd start-here-guidelines
git remote add upstream
https://github.com/zero-to-mastery/start-here-guidelines.git
git pull upstream master

Regards
Avinash

**Comment by avinnitesh** [comment]
Hi Anton,
I am getting below message as well for conflict:-
⚠️ *MERGE CONFLICT DETECTED!*

@avinnitesh <https://github.com/avinnitesh> A possible conflict has been
detected, you will need to resolve this before your pull request can be
merged. The most common reason conflicts occur, is when the contributor
does not run git pull origin master before pushing their new changes.

Before we can merge the code, you will need to resolve the conflict, check
out this video by Aldo: https://www.youtube.com/watch?v=uQLNFRviB6Q. If you
get stuck ask over on Discord.

—



I ran command and followed the step which came but could not able to
resolve the conflict:-

(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git pull origin master
remote: Enumerating objects: 34, done.
remote: Counting objects: 100% (34/34), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 32 (delta 16), reused 15 (delta 8), pack-reused 0
Unpacking objects: 100% (32/32), 7.61 KiB | 169.00 KiB/s, done.
From github.com:zero-to-mastery/start-here-guidelines
 * branch                master     -> FETCH_HEAD
   ac2d6ccf1..ef47335e6  master     -> origin/master
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint:
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint:
hint: You can replace "git config" with "git config --global" to set a
default
hint: preference for all repositories. You can also pass --rebase,
--no-rebase,
hint: or --ff-only on the command line to override the configured default
per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git config --global pull.rebase true
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git pull origin master
From github.com:zero-to-mastery/start-here-guidelines
 * branch                master     -> FETCH_HEAD
Auto-merging CONTRIBUTORS.md
CONFLICT (content): Merge conflict in CONTRIBUTORS.md
error: could not apply 53ed2160a... added my name
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git
rebase --abort".
Could not apply 53ed2160a... added my name
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git add CONTRIBUTORS.md
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git rebase --continue
[detached HEAD 95b343e95] added my name
 Committer: Avinash Kant ***@***.***>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly:

    git config --global user.name "Your Name"
    git config --global user.email ***@***.***

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 1 file changed, 4 insertions(+)
Successfully rebased and updated refs/heads/dev.
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant-> git rebase --continue
fatal: No rebase in progress?
(base) [~/Desktop/Project_github/start-here-guidelines]<strong>
</strong>avinash.kant->


Regards

Avinash

**Comment by avinnitesh** [comment]
Hi,
I have resolved this but my changes has not gone through yet as i had to
delete it as it is not sync with master.
Please let me know how to proceed.
Regards
Avinash
